Renderer.js

```diff
@@ -0,0 +1,88 @@
+class Renderer
```

Comment:
just having one renderer class with all the render methods would make it a lot cleaner instead of doing all the inheritance